File: python/algorithms/binary_search.py
def binary_search(sortedList, value):
    length = len(sortedList)

    if length == 1:
        return value == sortedList[0]

    mid = length / 2
    midValue = sortedList[mid]
    if value == midValue:
        return True
    elif value < midValue:
        return binary_search(sortedList[:mid], value)
    elif value > midValue:
        return binary_search(sortedList[mid:], value)


def main():
    pass


if __name__ == '__main__':
    main()

File: c++/src/algorithms/binary_search.cpp
#include "algorithms/binary_search.h"

bool binary_search(int find, int list[], int start, int length)
{
    int mid = (length / 2) + start;
    if (length == 1) {
        return find == list[start];
    } else if (find == list[mid]) {
        return true;
    } else if (find > list[mid]) {
        return binary_search(find, list, mid, (length + start) - mid);
    } else if (find < list[mid]) {
        return binary_search(find, list, start, mid - start);
    }
}